zcash_client_sqlite/src/wallet/init.rs

```diff
@@ -516,12 +520,17 @@ fn init_wallet_db_internal<P: consensus::Parameters + 'static>(
     Ok(())
 }
 
-/// Initialises the data database with the given [`UnifiedFullViewingKey`]s.
+/// Initialises the data database with the given set of account [`UnifiedFullViewingKey`]s.
+///
+/// **WARNING** This method should be used with care, and should ordinarily be unnecessary.
+/// Prefer to use [`zcash_client_backend::data_api::WalletWrite::create_account`] instead.
 ///
 /// The [`UnifiedFullViewingKey`]s are stored internally and used by other APIs such as
-/// [`get_address`], [`scan_cached_blocks`], and [`create_spend_to_address`]. `extfvks` **MUST**
-/// be arranged in account-order; that is, the [`UnifiedFullViewingKey`] for ZIP 32
-/// account `i` **MUST** be at `extfvks[i]`.
+/// [`get_address`], [`scan_cached_blocks`], and [`create_spend_to_address`]. Account identifiers
+/// in `keys` **MUST** form a consecutive sequence beginning at account 0, and the
```

Comment:
We should add a check for this (that the provided list's account IDs forms the appropriate set).

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/637/commits/51a14464a2697deabcf5a10718043e8476d2c306

Reply:
Fixed in https://github.com/zcash/librustzcash/pull/637/commits/55e3afdbcf3a499d1dd59d5c33104c0d4e0e9d2e